### PR TITLE
Add documentation for the Json Metadata file

### PR DESCRIPTION
**Pull request by austin0**
Adds some basic documentation for the Metadata .json file generated with every build.
Includes documentation for https://github.com/AdoptOpenJDK/openjdk-build/pull/2077 preemptively.

Signed-off-by: Austin Bailey <Austin.Bailey@ibm.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -316,16 +321,114 @@ Alongside the built assets a metadata file will be created with info about the b
...
+Example values: [`AdoptOpenJDK`, `Alibaba`]
+
+This tag is used to identify the vendor of the JDK being built, this value is set in the [build.sh](https://github.com/AdoptOpenJDK/openjdk-build/blob/805e76acbb8a994abc1fb4b7d582486d48117ee8/sbin/build.sh#L183) file and defaults to "AdoptOpenJDK".
+
+- `os:`
+Example values: [`windows`, `mac`, `linux`, `aix`, `solaris`]
+
+This tag identifies the operating system the JDK has been built on (and should be used on).
+
+- `arch:`
```
Comment (M-Davies) [inline README.md:346]: Missing explanation (to keep it the same as the others)

**Review by M-Davies — CHANGES_REQUESTED** [review]
One tinsy change. Otherwise, it looks good

File: README.md (hunk 1 above)
Comment (austin0) [reply README.md:346]: Added missing archs and a small description.

**Review by M-Davies — APPROVED** [review]

**Review by sxa — APPROVED** [review]
Thanks Austin - great start and we can fill in more details as we go on. I do wonder if it might be better having this information formatted as a table but I'm more interested in getting this in that worrying too much about the format for now, so I approve :-)

**Pull request merged**